squish/src/lib.rs

```diff
@@ -333,4 +333,60 @@ mod tests {
         let estimate = Format::Bc3.compressed_size(15, 30);
         assert_eq!(estimate, 512);
     }
+
+    // A colour test-pattern (RGB) with the first row in one colour,
+    // the second in another and the third and last row in a third colour.
+    static DECODED_BLOCK_COLOUR_4X4: &[u8] = &[
+        255, 150, 74, 255, 150, 74, 255, 150, 74, 255, 150, 74, // row 0
+        255, 120, 52, 255, 120, 52, 255, 120, 52, 255, 120, 52, // row 1
+        255, 105, 41, 255, 105, 41, 255, 105, 41, 255, 105, 41, // row 2
+        255, 105, 41, 255, 105, 41, 255, 105, 41, 255, 105, 41, // row 3
+    ];
+
+    // BC1 data created with AMD Compressonator v4.1.5083 and is the same as libsquish
+    static ENCODED_BLOCK_COLOUR_4X4: [u8; 8] = [0xA9, 0xFC, 0x45, 0xFB, 0x00, 0xFF, 0x55, 0x55];
+
+    fn decoded_block_colour_4x4_as_rgba() -> [u8; 4 * 4 * 4] {
+        let mut output = [0u8; 4 * 4 * 4];
+        for i in 0..4 * 4 {
+            output[i * 4 + 0] = DECODED_BLOCK_COLOUR_4X4[i * 3 + 0]; // R
+            output[i * 4 + 1] = DECODED_BLOCK_COLOUR_4X4[i * 3 + 1]; // G
+            output[i * 4 + 2] = DECODED_BLOCK_COLOUR_4X4[i * 3 + 2]; // B
+            output[i * 4 + 3] = 0xFF; //A
+        }
+        output
+    }
+
+    #[test]
+    fn test_bc1_decompression_colour() {
+        let encoded: [u8; 8] = ENCODED_BLOCK_COLOUR_4X4;
+        let mut output_actual = [0u8; 4 * 4 * 4];
+        Format::Bc1.decompress(&encoded, 4, 4, &mut output_actual);
+        assert_eq!(output_actual, decoded_block_colour_4x4_as_rgba());
+    }
+
+    #[test]
+    fn test_bc1_compression_colour_cluster_fit() {
```

Comment:
Misleading function name, the test actually tries all three algorithms. Also I'd rather have it describe the particular pattern or regression it tests.